handlers/twiml/handlers.go

```diff
@@ -274,62 +263,52 @@ func (h *handler) SendLegacy(ctx context.Context, msg courier.MsgOut, clog *cour
 		// build our URL
 		baseURL := h.baseURL(channel)
 		if baseURL == "" {
-			return nil, fmt.Errorf("missing base URL for %s channel", h.ChannelName())
+			return courier.ErrChannelConfig
 		}
 
 		sendURL, err := utils.AddURLPath(baseURL, "2010-04-01", "Accounts", accountSID, "Messages.json")
 		if err != nil {
-			return nil, err
+			return err
 		}
 
 		req, err := http.NewRequest(http.MethodPost, sendURL, strings.NewReader(form.Encode()))
 		if err != nil {
-			return nil, err
+			return err
 		}
 		req.SetBasicAuth(accountSID, accountToken)
 		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
 		req.Header.Set("Accept", "application/json")
 
 		resp, respBody, err := h.RequestHTTP(req, clog)
-		if err != nil {
-			return status, nil
+		if err != nil || resp.StatusCode/100 == 5 {
+			return courier.ErrConnectionFailed
 		}
 
 		// see if we can parse the error if we have one
 		if resp.StatusCode/100 != 2 && len(respBody) > 0 {
 			errorCode, _ := jsonparser.GetInt(respBody, "code")
 			if errorCode != 0 {
+				clog.Error(twilioError(errorCode))
 				if errorCode == errorStopped {
-					status.SetStatus(courier.MsgStatusFailed)
-
-					// create a stop channel event
-					channelEvent := h.Backend().NewChannelEvent(msg.Channel(), courier.EventTypeStopContact, msg.URN(), clog)
-					err = h.Backend().WriteChannelEvent(ctx, channelEvent, clog)
-					if err != nil {
-						return nil, err
-					}
+					return courier.ErrContactStopped
 				}
-				clog.Error(twilioError(errorCode))
-				return status, nil
+				return courier.ErrResponseUnexpected
 			}
+
+			return courier.ErrResponseStatus
 		}
 
 		// grab the external id
 		externalID, err := jsonparser.GetString(respBody, "sid")
 		if err != nil {
 			clog.Error(courier.ErrorResponseValueMissing("sid"))
-			return status, nil
+			return courier.ErrResponseUnexpected
```

Comment:
The send err types get converted to channel log errors... so the rule should be that you call `clog.Error` for non-terminal errors - i.e. something which is an error, but we'll carry on sending the message anyway